crates/sui-indexer-alt-framework/src/ingestion/error.rs

```diff
@@ -25,4 +25,7 @@ pub enum Error {
 
     #[error(transparent)]
     RpcClientError(#[from] tonic::Status),
+
+    #[error("Streaming error: {0}")]
+    StreamingError(#[from] anyhow::Error),
```

Comment:
This was my bad in the suggestion -- I think this should be:

```suggestion
    StreamingError(#[source] anyhow::Error),
```

Because `#[from]` creates a `From<anyhow::Error>` impl that is probably too general -- it makes it too easy for someone to use `into` or the `?` (try) operator on an `anyhow::Result` or `anyhow::Error` and wrap it in a `StreamingError` when it's not a streaming error.

`#[source]` is like `#[from]` but without that behaviour (requires manual wrapping with `map_err`).